### PR TITLE
Dblatcher/ar newsletter sign design variants

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?
(no public changes)
- adds an app-rendering Layout variant for the NewsletterSignup design (matching the design used in DCR)
- add variants for the Headline, Standfirst and Mainmedia components 
- creates a Layout story for the NewsletterSignup design, using a fixture with representative content

## Why?
We are planning to incrementally roll out the design on Apps rendering to keep the PR's manageable . 
See the design for further feature to added before putting the Layout live:
 - Newsletter data components (Frequency and region focus)
 - Share buttons
 - replacing the `ArticleBody` with a sign-up form component - if the client app supports the feature (being introduced in Bridget 2.0.0 - see https://github.com/guardian/dotcom-rendering/pull/6464)

## Screenshots

| Before      | After      | design brief |
|-------------|------------|--------|
| Screenshot 2022-11-28 at 16 11 02| Screenshot 2022-11-28 at 16 11 38| Screenshot 2022-11-28 at 16 13 59 |

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

**Review by georgeblahblah — COMMENTED** [review]
Thank you so much for this PR, David! It's looking great, and I think you've got the structure and approach to fit the patterns used in AR really nicely.

I had a few comments about some specific parts:

- In some cases, I wondered whether some of the component variants could also be served by the "Default" variant
- For the MainMedia component, I think it's best to show the caption/credit alongside the image. I'm not sure if this was specifically omitted in the designs, though
- The logo in the banner might need another look, specifically when the page font size is scaled
- The banner is potentially missing some dark mode styles. I think it's also worth taking a look at how this will look when shown in combination with the app's top banner, which is always present

One other suggestion I did have (which might be worth exploring together), is whether the [`grid.ts`](/apps-rendering/src/grid/grid.ts) module could be helpful in structing the layout. I'd be happy to pair with you on this and see where we can get to.

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:129]: For now, the article bodies include the old-style article embeds as a fall back for the AR version of the SignUp pages.

If we follow the method used on DCR, the Newsletter Sign Up component will be be put on the page instead of having an ArticleBody. This would need to wait until the AR Newsletter sign up component was full functional.

The SignUp pages have the same tags used to place the in in-article sign-up blocks, so the MAPI data for them should include the `PromotedNewsletter` meta data that can be used to set the props for a Newsletter sign up component within the layout.

**Comment by dblatcher** [comment]
Thanks for the feedback on this @georgeblahblah  - we've got some more UX / Design work coming for sign up pages in the App, so it might be best to pick up on this branch when we have that to work off.

File: apps-rendering/src/fixtures/newsletterSignUpContent.ts
Comment (dblatcher) [reply apps-rendering/src/fixtures/newsletterSignUpContent.ts:80]: Yes, the idea will be to transition to those components and not display an articleBody (like on the DCR version). We are  further along with making those work no, but I think its best to leave that to a later PR - especially as we'll have to consider what to display on Client Apps with a Bridget version that won't support them.

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:43]: 👍

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:72]: Thanks - The revised design for app doen't include this banner so I'll remove it altogether.

File: apps-rendering/src/components/Headline/NewsletterSignupHeadline.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Headline/NewsletterSignupHeadline.tsx:30]: I think I just interpreted the pattern to be each design having its own Headline file - i'll change it to use `DefaultHeadline`

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:90]: 👍

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:124]: Thanks - we do have one sponsored newsletter so far, but it doesn't use that kind of logo - the branding is part of the graphic: https://www.theguardian.com/lifeandstyle/2022/sep/28/sign-up-for-the-house-to-home-newsletter
It's probably best to remove the `Logo` from this layout until we have a design brief for how they should be incorporated (if at all)

File: apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx
Comment (dblatcher) [reply apps-rendering/src/components/Layout/NewsletterSignUpLayout.tsx:17]: thanks - I hadn't noticed it was from editions - i'll use a value from `breakpoints` instead.

**Comment by dblatcher** [comment]
> One other suggestion I did have (which might be worth exploring together), is whether the [`grid.ts`](https://github.com/guardian/dotcom-rendering/blob/main/apps-rendering/src/grid/grid.ts) module could be helpful in structing the layout. I'd be happy to pair with you on this and see where we can get to.

I've given this a try - it's looking a lot cleaner using `grid`!

**Review by georgeblahblah — APPROVED** [review]
Looks great! Thanks David

**Pull request merged**
